autoPyTorch/pipeline/components/base_choice.py

```diff
@@ -247,35 +249,37 @@ def _check_dataset_properties(self, dataset_properties: Dict[str, Any]) -> None:
         """
         assert isinstance(dataset_properties, dict), "dataset_properties must be a dictionary"
 
-    def _apply_search_space_update(self, name: str, new_value_range: Union[List, Tuple],
-                                   default_value: Union[int, float, str], log: bool = False) -> None:
+    def _apply_search_space_update(self, hyperparameter_search_space_update: HyperparameterSearchSpaceUpdate) -> None:
         """Allows the user to update a hyperparameter
 
-        Arguments:
+        Args:
             name {string} -- name of hyperparameter
             new_value_range {List[?] -- value range can be either lower, upper or a list of possible conditionals
             log {bool} -- is hyperparameter logscale
         """
 
-        if len(new_value_range) == 0:
-            raise ValueError("The new value range needs at least one value")
-        self._cs_updates[name] = tuple([new_value_range, default_value, log])
+        self._cs_updates[hyperparameter_search_space_update.hyperparameter] = hyperparameter_search_space_update
 
-    def _get_search_space_updates(self, prefix: Optional[str] = None) -> Dict[str, Tuple]:
+    def _get_search_space_updates(self, prefix: Optional[str] = None) -> Dict[str, HyperparameterSearchSpace]:
         """Get the search space updates with the given prefix
 
-        Keyword Arguments:
+        Args:
             prefix {str} -- Only return search space updates with given prefix (default: {None})
 
         Returns:
             dict -- Mapping of search space updates. Keys don't contain the prefix.
         """
+        RETURN_ALL = False
         if prefix is None:
-            return self._cs_updates
-        result: Dict[str, Tuple] = dict()
+            RETURN_ALL = True
+
+        result: Dict[str, HyperparameterSearchSpace] = dict()
 
-        # iterate over all search space updates of this node and filter the ones out, that have the given prefix
+        # iterate over all search space updates of this node and keep the ones that have the given prefix
         for key in self._cs_updates.keys():
-            if key.startswith(prefix):
-                result[key[len(prefix) + 1:]] = self._cs_updates[key]
+            if RETURN_ALL:
+                result[key] = self._cs_updates[key].get_search_space()
+            elif re.search(f'^{prefix}', key) is not None:
```

Comment:
Actually, I think you would not need the assert if you do the change of `if prefix is None` in this for loop

Reply:
True